### ISSUE TITLE
FR: Allow to specify exit-node for certain applications only on Android

### ISSUE THREAD

**Issue by amosbird** [tailscale/tailscale#4971]
### What are you trying to do?

I'm trying to use tailscale and enable exit-node for certain applications only. It's not ideal to use exit-node globally.

### How should we solve this?

Android has some mechanism to do per-app routing, which can be utilized as a per-app exit-node configuration.

### What is the impact of not solving this?

Without such feature, I cannot make tailescale always on.

**Comment by DentonGentry** [comment]
Could I ask the actual problem you are trying to solve? This is proposing a solution, the problem is that you need to use the VPN for only some public destinations?

Are you looking for a way to have an known public IP address to add to some service's Allowlist?

**Comment by amosbird** [comment]
> the problem is that you need to use the VPN for only some public destinations?

Nope. I'm looking for a way to use the VPN for certain applications only, not destinations.

**Comment by wizzard0** [comment]
Chiming in: would be really cool if this is possible to do on iOS, too. Exclude some apps.

Scenario: Some finance apps like to throw a tantrum when they see country suddenly change underneath, and it’s quite a nuisance to force-quit all of them before using an exit node, and avoid accidentally launching them while an exit node is active

**Comment by Renari** [comment]
Would be nice to have this on other platforms as well, many VPN services have functionality to tunnel single applications through an exit node.

**Comment by GusevAG** [comment]
same for me
I have exit node with a static IP on a VPS, through which I have to access my "working space" with a specific program.
this program uses very little traffic, but very sensitive to ping.
while routing all the traffic through the exit node, I add to the ping, and it would be nice to route just that specific app to that exit node.
